### PR TITLE
Vendor links

### PR DESCRIPTION
**Pull request by LizzAlice**
solves #221, but without dell api

### CODE REVIEW

File: devices/models.py [hunk 1]
```diff
@@ -100,6 +102,35 @@ class Bookmark(models.Model):
     user = models.ForeignKey(Lageruser)
 
 
+class Vendor(models.Model):
+    name = models.CharField(_('Name'), max_length=100)
+    driver_url = models.CharField(_('Drivers'), max_length=1000, blank=True, null=True)
```
Comment (octomike) [inline devices/models.py]: why not `model.URLField` ?

File: devices/templatetags/devicetags.py [hunk 2]
```diff
@@ -129,3 +129,15 @@ def as_nested_list(factvalue):
 @register.filter
 def splitstr(arg1, arg2):
     return arg1.split(arg2)
+
+
+@register.filter
+def addstr(arg1, arg2):
```
Comment (octomike) [inline devices/templatetags/devicetags.py]: this not supposed to be part of the PR, right?

File: devices/models.py (hunk 1 above)
Comment (octomike) [reply devices/models.py]: also, what's the difference between Vendor and Manufacturer?

File: devices/models.py [hunk 3]
```diff
@@ -100,6 +102,35 @@ class Bookmark(models.Model):
...
+        return reverse('vendor-detail', kwargs={'pk': self.pk})
+
+    def get_edit_url(self):
+        return reverse('vendor-edit', kwargs={'pk': self.pk})
+
+    def get_associated_devices(self):
+        return Device.objects.filter(vendor=self)
+
+    def clean(self):
+        if "SERVICETAG" not in self.driver_url:
```
Comment (octomike) [inline devices/models.py]: The name "SERVICETAG" is a specific name for a device property of a specific manufacturer/vendor (Dell).
Ideally this needs to be dynamically added to each URL. Some vendor links may require the serial number, some the exact product name and so on.

Iff you decided to hard code these properties, I think they should at least be given the generic name (here: serialnumber).

File: devices/views.py [hunk 4]
```diff
@@ -967,6 +967,78 @@ def post(self, request, **kwargs):
         return HttpResponseRedirect(reverse("device-detail", kwargs={"pk": device.pk}))
 
 
+class VendorList(PaginationMixin, ListView):
```
Comment (octomike) [inline devices/views.py]: Again - why don't you use Manufacturer?

File: devices/models.py [hunk 5]
```diff
@@ -100,6 +102,35 @@ class Bookmark(models.Model):
     user = models.ForeignKey(Lageruser)
 
 
+class Vendor(models.Model):
+    name = models.CharField(_('Name'), max_length=100)
+    driver_url = models.CharField(_('Drivers'), max_length=1000, blank=True, null=True)
+    support_url = models.CharField(_('Support'), max_length=1000, blank=True, null=True)
```
Comment (octomike) [inline devices/models.py]: Also, I was thinking about a more generic approach to support URLs. I mean, they could be totally different for each vendor. A number of generic URL templates would be much more flexible. This way a user could add a support and a drivers URL with a URL template for Dell and a single selfservice URL template for Apple.

File: devices/models.py (hunk 1 above)
Comment (LizzAlice) [reply devices/models.py]: I didn't notice that there was already a manufacturer :S ok, changing vendor -> manufacturer

File: devices/templatetags/devicetags.py (hunk 2 above)
Comment (LizzAlice) [reply devices/templatetags/devicetags.py]: Yes it is, I wanted a template tag that doesn't change behavior when given a string that can be cast to an int and as I first needed it in this pr, I thought it belonged in this pr.

**Pull request closed without merging**